homeassistant/components/coinbase/config_flow.py

```diff
@@ -0,0 +1,148 @@
+"""Config flow for Coinbase integration."""
+import logging
+
+from coinbase.wallet.client import Client
+from coinbase.wallet.error import AuthenticationError
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_API_KEY, CONF_API_TOKEN
+
+from .const import (  # pylint:disable=unused-import
+    API_ACCOUNT_CURRENCY,
+    API_ACCOUNTS_DATA,
+    API_RATES,
+    CONF_CURRENCIES,
+    CONF_EXCAHNGE_RATES,
+    CONF_YAML_API_KEY,
+    CONF_YAML_API_TOKEN,
+    CONF_YAML_CURRENCIES,
+    CONF_YAML_EXCHANGE_RATES,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Required(CONF_API_TOKEN): str,
+        vol.Optional(CONF_CURRENCIES): str,
+        vol.Optional(CONF_EXCAHNGE_RATES): str,
```

Comment:
I realize the number of currencies and exchange rates available might be problematic, but asking for comma-separated, free form, text input in this case feels weird as well.

Should this be a multi-select? And should it be part of the main configuration flow? Or should this be an option flow instead? (So a user can adjust later on).

Reply:
It 100% should be an option flow IMO. I was planning to do that in a separate PR down the line but perhaps it should be here to avoid a breaking change for users who set up via config flow?

Reply:
Actually the options flow is looking pretty good. Nearly there with it just need to add some validation and tests